uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -726,135 +757,112 @@ define([
     }
 
-    /**
-     * Converts given date to javascript date as expected by uib-datepicker
-     *
-     * @param {String} date from server
-     * @return {Date}
-     */
-    function _convertDateFormatFromServer (date) {
-      return moment(date, sharedSettings.serverDateFormat).toDate();
-    }
-
-    /**
-     * Converts given date to server format
-     *
-     * @param {Date} date
-     * @return {String} date converted to server format
-     */
-    function _convertDateToServerFormat (date) {
-      return moment(date).format(sharedSettings.serverDateFormat);
-    }
-
-    /**
-     * Initializes user's calendar (work patterns)
-     *
-     * @return {Promise}
-     */
-    function _loadCalendar () {
-      return Calendar.get(vm.request.contact_id, vm.period.start_date, vm.period.end_date)
-        .then(function (usersCalendar) {
-          vm.calendar = usersCalendar;
-        });
-    }
-
-    /**
-     * Sets dates (and times in case of "hours" absence type) for vm.request from UI
-     */
-    function _setDates () {
+    function setDatesToRequest () {
       var options = vm.uiOptions;
       var request = vm.request;
-      var times;
+      var times = options.times;
 
-      request.from_date = options.fromDate ? vm._convertDateToServerFormat(options.fromDate) : null;
-      request.to_date = options.toDate ? vm._convertDateToServerFormat(options.toDate) : null;
+      request.from_date = options.fromDate ? convertDateToServerFormat(options.fromDate) : null;
+      request.to_date = options.toDate ? convertDateToServerFormat(options.toDate) : null;
 
       if (isCalculationUnit('hours') && !isLeaveType('toil')) {
-        times = options.times;
         request.from_date = request.from_date && times.from.time ? request.from_date + ' ' + times.from.time : null;
         request.to_date = request.to_date && times.to.time ? request.to_date + ' ' + times.to.time : null;
-        request.from_date_amount = !isNaN(+times.from.amount) ? times.from.amount : null;
-        request.to_date_amount = !isNaN(+times.to.amount) ? times.to.amount : null;
       }
 
       if (!options.multipleDays && options.fromDate) {
-        options.toDate = options.fromDate;
         request.to_date = request.from_date;
+        request.to_date_type = request.from_date_type;
+        options.toDate = options.fromDate;
       }
     }
 
     /**
-     * Sets dates and types for vm.request from UI
+     * Sets dates (with time in case of "hours" Absence Type) from UI to vm.request
+     *
+     * @param  {String} dateType from|to
+     * @return {Promise}
      */
-    function _setDateAndTypes () {
-      vm._setDates();
+    function setDatesFromUI (dateType) {
+      resetAfterDateChange(dateType, true);
+      setDatesToRequest();
 
-      if (!vm.uiOptions.multipleDays && vm.uiOptions.fromDate) {
-        vm.request.to_date_type = vm.request.from_date_type;
-      }
+      return $q.resolve()
```

Comment:
while you're at it...
```js
function setDatesFromUI (dateType) {
  return $q.resolve()
    .then(function () {
      resetAfterDateChange(dateType, true);
    })
    .then(setDatesToRequest);
    .then(vm.setDatesFromUIExtended ? vm.setDatesFromUIExtended : _.noop)
    .then(function () {
      return checkAbsencePeriod(dateType);
    });
```

Reply:
Not sure I understand what do you mean.

Reply:
@AkA84 I think this comment is completely irrelevant after changes made on the all comments above...